synapse/handlers/sync.py

```diff
@@ -2116,7 +2147,24 @@ async def _get_rooms_changed(
             room_entry = room_to_events.get(room_id, None)
 
             newly_joined = room_id in newly_joined_rooms
-            if room_entry:
+
+            # In case of a non lazy-loading-members sync we want to include
+            # rooms that got un partial stated in the meantime, and we need
+            # to include the full state of them.
+            if (
+                not sync_config.filter_collection.lazy_load_members()
+                and room_id in un_partial_stated_rooms
+            ):
+                entry = RoomSyncResultBuilder(
+                    room_id=room_id,
+                    rtype="joined",
+                    events=None,
```

Comment:
Is this correct? In the `else` branch below we pass `events=[]`. Is there any difference between passing `None` and an empty list here?

Reply:
Yes and no :)

`events=[]` seems to be a [shortcut](https://github.com/matrix-org/synapse/blob/develop/synapse/handlers/sync.py#L2281) for `no changes`. In our case we bypass the shortcut because we have `full_state=True`.

`events=None` also has a specific case further down in the call stack, cf `potential_recents` in `_load_filtered_recents`. It seems to be harmless when `newly_joined=True` however.

I'll be frank, I am not completely sure about what is the purpose of this parameter.